### PR TITLE
Updated package Deps

### PR DESCRIPTION
**Pull request by Toyz**
- Added **.idea** to the _.gitignore_
- Updated cli to cli/v2 (This uses go.mods for versioning)

### CODE REVIEW

**Comment by Toyz** [comment]
Refactored some parts of the config code in this PR as well...

- Removed not needed pointer in hot path
- Reformatted the file to put unexported on the bottom of the file
- Write will now write "sane" JSON so a user can easily edit it

-- Note if your base struct is a pointer you do not need nested pointers as that is useless... You can still modify the entire struct even if the item isn't a point as the entire struct is tracked

**Comment by Toyz** [comment]
Closing this

File: config.go [hunk 1]
```diff
@@ -87,15 +81,15 @@ func CreateConfig(configPath, repo string) *Config {
...
 		basePath:      basePath,
 		localRepoPath: localRepoPath,
-		vals:          &ConfigSpec{&repo, false, false},
+		vals:          &ConfigSpec{repo, false, false},
 	}
 }
 
 func (c *Config) Repo() string {
-	if c.vals.Repo == nil {
+	if strings.EqualFold(c.vals.Repo, "") {
```
Comment (voldyman) [inline config.go:89]: how is `strings.EqualFold` more helpful than `len(c.vals.Repo) == 0`

File: config.go [hunk 2]
```diff
@@ -115,3 +109,9 @@ func (c *Config) Write(path string) error {
 
 	return ioutil.WriteFile(path, data, os.ModePerm)
 }
+
+/* unexported */
+func parseConfig(cfg []byte) (spec *ConfigSpec, err error) {
```
Comment (voldyman) [inline config.go:114]: nit: i personally prefer not having named return types but i'll leave it to @ThePrimeagen to make the last call on it.

File: config.go [hunk 3]
```diff
@@ -16,7 +17,7 @@ var (
 )
 
 type ConfigSpec struct {
-	Repo            *string `json:"repo"`
+	Repo            string `json:"repo"`
```
Comment (voldyman) [inline config.go:20]: no `gofmt`?

**Comment by Toyz** [comment]
I closed this in favor of multi-repo, I'll move them to the other branch

**Pull request closed without merging**